src/package.coffee

```diff
@@ -590,10 +590,21 @@ class Package
       false
 
   # Get an array of all the native modules that this package depends on.
-  # This will recurse through all dependencies.
+  #
+  # First try to get this information from
+  # @metadata._atomModuleCache.extensions. If @metadata._atomModuleCache doesn't
+  # exist, recurse through all dependencies.
   getNativeModuleDependencyPaths: ->
     nativeModulePaths = []
 
+    if @metadata._atomModuleCache?
+      nativeModuleBindingPaths = @metadata._atomModuleCache.extensions?['.node'] ?[]
+      for nativeModuleBindingPath in nativeModuleBindingPaths
+        # The `.node` file lies in nativeModulePath/build/Release/ folder.
+        nativeModulePath = path.normalize(path.join(path.dirname(nativeModuleBindingPath), '..', '..'))
```

Comment:
I believe `path.join` normalizes the result so you don't need an additional call to `path.normalize`